src/block_ffm.rs

```diff
@@ -3,6 +3,7 @@ use std::any::Any;
 use std::error::Error;
 use std::io;
 use std::mem::{self, MaybeUninit};
+use std::simd::{f32x4, SimdFloat};
```

Comment:
cool!
1) did you see a speedup for a FFM only model? (I would measure first just for inference). also might be better to measure this with your "send context once" improvement so that any potential bottleneck in memory fetches is diminished a bit.
2) why f32x4? (just curious). I see [here](https://doc.rust-lang.org/src/core/portable-simd/crates/core_simd/src/vector/float.rs.html#15) they also have f32x16, (assuming it's backed by AVX512 but didn't check :P)

Reply:
1. I did see a speed up also with only the SIMD, not as pronounced as with the "send context once" improvement
2. In constants that we base our iterations on top of (ffm_k etc) are normally either smaller than 16 or would for 8 actually cause us to do more iterations than with 4. Since there also wasn't a performance gain with f32x8 I picked the middle ground (f32x4)